Automattic/wp-calypso
# Update Jetpack Cloud Download Styling

#### Changes proposed in this Pull Request

* Add new Jetpack Cloud `RewindFlow` component to be used for both restore and download flow
* Implement the download half of `RewindFlow`
* Remove all existing download-flow specific components
* Add and refine download related selectors to clarify flow

![The Download Flow](https://user-images.githubusercontent.com/2810519/77355085-13a1ff00-6d01-11ea-8dd4-65b0f53d3504.png)


#### Testing instructions

1. Navigate to `/backups/` for a site that has Jetpack backups
2. Pick a backup to download, and click "Download Backup"
3. Verify you are on a page that matched "1. Confirm & Configure Download" above
4. Make at least one change to the download configuration, noting what it is, and open network tools.
5. Click "Create downloadable file"
6. Find the POST to `/wpcom/v2/sites/:sitId/rewind/downloads` in the network tools and verify that the body matched the configuration demo step 4.
7. Verify the in-progress screen matches "2. Download In-Progress" above
8. Verify that eventually the screen goes to "3. Download Ready"
9. Verify that the download is ready
10. Verify you received an email for the site saying the download is ready
11. Reload the page and verify after briefly seeing "1. Confirm & Configure Download" the page jumps to "3. Download Ready", since the download is already ready.

#### Next Steps

1. Better error handling. Anything in the flow that is not expected will result in this screen:
<img width="300" src="https://user-images.githubusercontent.com/2810519/77355613-f6216500-6d01-11ea-8978-d7ab9fa0bad7.png" >

2. Bring the restore flow in using the `RewindFlow` component to reduce repeated style code and components.

3. Rework of the underlying request for downloads. It returns 0-10 of the most recent backups, but we only grab the first if it exists. Handling for zero returned downloads is poor. We would want to
    a. Get all returned backups to reduce re-doing download creation
    b. improve the handling of 0 returned downloads. This will also allow request progress to avoid the "jump" from step 11
